modules/ROOT/pages/6.4.1-release-notes.adoc

```diff
@@ -0,0 +1,1334 @@
+
+Improvements were made to check if the DOM parent node is a `contenteditable="false"` element.
+
+In {productname} 6.4.1, elements with `contenteditable="false"` attributes can no longer be selected when they are children of another `contenteditable="false"` attribute, such as:
+
+.Example of `contenteditable="false"` elements inside `contenteditable="false"` root.
+[source,HTML]
+----
+<div> Parent div element
+  <div contenteditable="false">First child noneditable div element</div>
+  <div contenteditable="true">Second child editable div element</div>
+</div>
+----
+
+=== Ranged deletion of formatted text using selection or keyboard shortcut would sometimes cause Blink and Webkit browsers to insert interpreted tags upon typing, which may result in inconsistent tags
+//#TINY-9302
+
+Blink & Webkit browsers (Chrome & Safari) were deleting the contents of inline formats which would cause the format to be lost and not retained. As a consequence, this issue would impact the semantics and visuals for the *HTML* document, which is not be visible to the user.
+
+A override fix already exists for _character-by-character_ deletion using backspace/delete keys with a collapsed selection. However, this particular bug persists when the format is deleted when there is a ranged selection that starts at the beginning of the inline format(s) and encompasses at least the entirety of the inline format(s). The bug also occurs when deletion is done using common keyboard shortcuts including `Ctrl/Alt + Backspace`, `Ctrl/Alt + Delete`, and `Cmd + Backspace`.
+
+In {productname} 6.4.1, two new key overrides have been added to:
+
+. Check that a `non-collapsed` selection satisfies the conditions set (specifically for a ranged selection using `backspace/delete`).
+. Check a range of conditions *before* continuing (for keyboard shortcut `backspace/delete`).
+
+As a result, the ranged selection for backspace/delete now check that all of the original formats at the cursor selection *start*, before deletion, are retained with either the native browser behaviour or a caret. For keyboard shortcuts that are using backspace/delete, will create a new unformatted caret which prevents browsers from inserting interpolated formats.
+
+=== Visual characters were rendered inside noneditable elements
+//#TINY-9474
+
+In previous versions of {productname}, a bug was found that affected the `visual characters` plugin. As a consequence, the visual characters selector would render inside `contenteditable="false"` elements after the plugin was applied to the editor.
+
+To fix this issue, a conditional filter was added which identifies, if an element is either a `contenteditable="true"` element or a `contenteditable="false"` element to its decendants.  If the element matches as a `contenteditable="false"` element to itself or decendants, the `visual character` plugin will remove the highlighted selecter from within the `dom` viewport.
```

Comment:
```suggestion
In order to address this problem, improvements were made that now detects whether an element or any of its descendants has the `contenteditable="true"` or `contenteditable="false"` attribute. If the element is identified as a `contenteditable="false"` element, either by itself or through its descendants, the visual character plugin will eliminate the highlighted selector from the dom viewport.
```

Reply:
@tiny-ben-tran reposting this as the old suggestion was outdated.

Reply:
```suggestion
In order to address this problem, improvements were made to better detect noneditable context, when an element is identified as a `contenteditable="false"` element, either by itself or through its descendants.

```